### DISCUSSION TITLE
next activity name

### DISCUSSION THREAD

**Discussion by rmanjusha** [elsa-workflows/elsa-core#5489] [Q&A]
Is it possible to return next activity name based on outcome  without executing it in using Elsa?

**Comment by glime-ai** [comment]
Based on the context provided, Elsa workflows support conditional branching where you can define outcomes that determine the next activity to execute. However, if you want to determine the next activity's name based on an outcome without actually executing it, you would need to inspect the workflow definition and the current state of the workflow instance programmatically.

Elsa does not provide a built-in feature to simply return the name of the next activity based on an outcome without executing it. You would need to write custom code to analyze the workflow definition and determine the next activity that would be executed for a given outcome.

For example, you could access the workflow definition, find the current activity, inspect its outcomes, and then determine which activity is linked to the outcome you're interested in. This would require a good understanding of the Elsa workflow model and possibly direct interaction with the workflow persistence store to fetch the necessary information.

If you need to implement such a feature, you might want to look into the Elsa source code or documentation to understand how workflows are executed and how you can interact with the workflow definitions and instances programmatically.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
